### PR TITLE
New skidding particle effect (and some various small fixes)

### PR DESCRIPTION
**Pull request by qwertychouskie**
The textures are here:

https://jacobspctuneup.tk/STK/new-skid-effect/skid-particle1.png
https://jacobspctuneup.tk/STK/new-skid-effect/skid-particle2.png

### CODE REVIEW

**Comment by qwertychouskie** [comment]
Some (grainy) videos:

https://jacobspctuneup.tk/STK/newskidparticlesA.webm
https://jacobspctuneup.tk/STK/newskidparticlesB.webm

**Comment by auriamg** [comment]
maybe, it doesn't seem all that different from what we already have?
it also seems like you're emitting way more particles than now, this could have performance implications

**Comment by qwertychouskie** [comment]
Here is a video of the old effect for comparison:

https://jacobspctuneup.tk/STK/oldskidparticles.webm

Notice how the particles are rigidly attached to the back of the kart and thus feel unnatural.

**Comment by auriamg** [comment]
I'd like opinions from other people, but TBH I still think the current effect looks better, the new proposed effect is just a thin yellow line

**Comment by hiker** [comment]
+1 to auria: in the first (new) movie the particles look like they are going more down than backwards, and the second one does imho not look as good as ours.

**Comment by qwertychouskie** [comment]
I made some more tweaks (and a small fix for nitro), testing is appreciated, I'm also thinking about maybe changing the PNGs.

**Comment by auriamg** [comment]
Screenshots/videos would be appreciated to help evaluate

**Comment by qwertychouskie** [comment]
Updated video: https://jacobspctuneup.tk/STK/new-skid-effect/demo.mkv

**Comment by auriamg** [comment]
Those look much better than the previous attempt, though I'd like other opinions from other people please?

**Comment by auriamg** [comment]
One thing missing is the license for the images

**Comment by qwertychouskie** [comment]
Oops... CC0

File: src/karts/kart_gfx.cpp [hunk 1]
```diff
@@ -234,15 +238,23 @@ void KartGFX::reset()
...
  */
 void KartGFX::setSkidLevel(const unsigned int level)
 {
-    assert(level >= 1);
+    assert(level >= 0);
     assert(level <= 2);
     m_skid_level = level;
-    const ParticleKind *pk = level==1 ? m_skid_kind1 : m_skid_kind2;
+    const ParticleKind *pk;
+    if (level == 0) {
```
Comment (auriamg) [inline src/karts/kart_gfx.cpp]: Sorry I missed that. This is not indented properly, please follow our conventions next time

**Comment by nado** [comment]
You might also consider git rebase next time, "Final?" is not much explicit as a commit message.

**Pull request merged**